### DISCUSSION TITLE
🐕 Pets

### DISCUSSION THREAD

**Discussion by adamhannigan** [sunflower-land/sunflower-land#2202] [Art]
Sunflower Land is about growing crops and raising livestock, but we believe adding a pets feature will increase player engagement and add another layer of fun to the game. Players will be able to collect dogs, pigs, and goats, and take care of them to increase their farming abilities.

The objective of the pets feature is to provide players with a fun and engaging way to increase their farming abilities. By feeding and caring for their pets, players will see improvements in their crop yields and livestock production.

<img width="561" alt="Screen Shot 2023-02-08 at 6 49 32 pm" src="https://user-images.githubusercontent.com/11745561/217467130-b59a5e12-7451-4494-98a3-4a54d6a21cf0.png">

**Abilities**

Players will be able to collect dogs, pigs, and goats, each with its unique abilities. For example, dogs can help guard livestock, pigs can help with tilling soil, and goats can help with clearing weeds.

Dog +0.2 Livestock Yield
Pig 20% Faster Crops
Goat +0.2 Crop Yield

**Pet Care**

To keep your pets happy and increase their abilities, players will need to feed and care for them on a regular basis. They can do this by providing food or attention.

Once a pet is satisfied with love and food its buff will become active for **24 hours.** Each pet will have unique tastes that require different types of food to satisfy.

## How to attain a Pet?

A pet is an advanced concept that will become available at Helios during certain levels.

Pig - Level 12
Dog - Level 20
Goat - Level 32

A player can have all 3 pets on their farm providing boosts.

**Pet Variations**

We aim to release collectible versions of pets that players can craft through quests & special events. By combining your average pet with exotic ingredients you will unlock a legendary version.

These variations will be seasonal and/or during special events. These pets may be cosmetic and/or provide a bonus buff.

**Pet Jealousy**

You can have multiple of the same pet, however you can only feed one each day. Pets are extremely jealous and ignore you if you provide attention to another one of their kind.

**Comment by SarielWang93** [comment]
Better distinguish between dog and Farm Dog, may be misunderstood.

**Comment by rv-02** [comment]
Growing up playing harvest moon, I can't think of anything more exciting than taking care of animals/pets in a farm simulation game. Love the feeding/caring daily part that encourages more game play.

On the **abilities and boosts** part, '20% faster crops' sounds a lil too much to me. At the moment we have the following crop harvest speed boost and they are stackable:

1. Legacy - Speed Specialist (10% quicker)
2. Cultivator (5% quicker)
3. Scarecrow (15% quicker)
4. Lunar Calendar (10% quicker)

The above makes at 24h crop reduces to 15h18m, adding 20% more will make it around 12h15m, basically half of original CD.
If it's thought through and intended that way (with more sinks introduced in the upcoming features) then that's cool, otherwise I thought maybe dial it back a lil.

On the other hand IMO based on its perceived value the progression should be in this order:
Level 12: +0.2 crop yield
Level 20: +0.2 livestock yield
Level 32: TBD% faster crop

Lastly, dog definitely feels right to be categorized by pet' but aren't pig and goat more like livestocks? Hmm.

**Comment by mmaxxpeinn** [comment]
Are the SFTs in this one open for art contribution from community artists or it will be assign to the team artists? ☺️

**Reply by adamhannigan** [reply]
@mmaxxpeinn  To be decided! There are going to be lots of variations so we will need lots of help.

We may get the team artists to design 3 base versions (e.g. dog, pig, goat)

The community then creates the rare/legendary ones ⭐

**Comment by TheDcol** [comment]
Really cool and exciting suggestions! Cannot wait :D

My only 'criticism' would be that we need to provide the demand for the resources players create. More boosts = more resources = oversupply risk. We must ensure there is a demand and even an ingame burn/sink mechanic to balance out all the boosts that are coming out. Leaning more towards main island sinks vs optional islands/events sinks

Excited to have all 3 and nurture them

**Reply by adamhannigan** [reply]
Definitely! @TheDcol 

It could be worth investigating some cross resource balancing. For instance, you must give a Dog sticks to play with. This burns the equivalent wood SFL that would be gained by earning crops SFL.

We will also incorporate pets into the baseline for what players can earn. For instance, we will assume that players level 20+ will have a dog and hence make expansions + buildings at those levels cost 10% more in resources

**Comment by rodolfoddias** [comment]
I love this concept and It's really warming to see other fellows that, just like me, grew up playing Harvest Moon. 
@adamhannigan have you guys considered adding a leveling system to the pets too? This way the buffs could scale not only based on players pet caring history.

Considering new pet releases from the same type, we could relate it to how good you took care from previous ones. For example:

Lilly is a Veterinarian Bumpkin from Sunflower Isles. You can reach her to obtain new pets as long as you reach certain conditions, such as:
- Bumpkin Level
- Farmer achievements (you must prove you're a hard working Bumpkin, not only some lazy one that keeps digging holes for cake's experience)
- Animal Care certificates, based on how good you take care of your current pets

The first dog could be obtained based only on Bumpkin Level + Farmer Achievements. However, in order to unlock a Tier 2 Farm Dog, you must maximize your previous one. Example:

- Tier 1 Dog: +0,1 Livestock / Max Level 10 
- Tier 2 Dog: +0,2 Livestock / Max Level 15 
- Etc...

Considering this system, the expected loop would be:
1) Player level up and reach required farmer achievements
2) Player buys Tier 1 Dog from Lilly
3) Player feeds the Tier 1 Dog a special treat to turn the daily buff on
4) Player feeds the Tier 1 regular food to increase its bond and level
5) Once the Tier 1 dog reaches max level, player goes to Lilly and buys Tier 2 dog
6) Player restarts the loop with Tier 2 dog and Tier 1 dog stays as a decoration / collectible (since they will probably activate buff from highest tier pet)
7) The same loop repeats for next Dog Tier (or any other animal)

OBS: The farmer achievements needed would also scale as a condition to obtain new pets. 

This way we can still attract players with some nice and juicy buffs, but they will have to work on them by taking care of multiple pets to prove themselves worthy to Lilly.

**Reply by adamhannigan** [reply]
@rodolfoddias That is an interesting idea! I would love to level up my pets.

As a whole, this feels like a right fit for pets since players will develop attachments to them.

Right now I do foresee a technical limitation. To enable players to withdraw and trade the pets, we would need to launch each pet as a unique NFT and store a bunch of other data on chain. This ensures that when pets are traded, their level and other attributes move with it.

Great idea! We just need some brain power to figure out the implementation for this 😅

**Comment by Sunflowerfarmer** [comment]
Is the DOG mentioned here is farm dog nft? If yes then what about "heard sheep 4x faster" that's like really powerful buff and many people are sitting on farm dog for a long time now. 🙂

**Reply by adamhannigan** [reply]
That is a great question!

Right now, there are no plans for Sheep in Sunflower Land. It was a community NFT from Sunflower Farmers that was carried over.

We can ask the dog holders what they would prefer:
- Integrate their Dog into the Pet Proposal with some added utility
- Do not integrate the dog and wait for sheep (no plans yet)

What do you think should happen?
